protocol/tags.go

```diff
@@ -87,7 +87,7 @@ const TopicMsgRespTagMaxSize = 6 * 1024 * 1024
 // TxnTagMaxSize is the maximum size of a TxnTag message. This is equal to SignedTxnMaxSize()
 // which is size of just a single message containing maximum Stateproof. Since Stateproof
 // transactions can't be batched we don't need to multiply by MaxTxnBatchSize.
-const TxnTagMaxSize = 4620031
+const TxnTagMaxSize = 8460031
```

Comment:
Hopefully we will not need to change this once we compute the maximum logicsig size correctly.

Reply:
I don't think we can get back to the original value now that we are changing the value of `MaxLogicSigMaxSize` from 1K to 16K.
The current formula is:
```
s = 1 + 2 + msgp.BytesPrefixSize + config.MaxLogicSigMaxSize + 4 + crypto.SignatureMaxSize() + 5 + crypto.MultisigSigMaxSize() + 4
s += msgp.ArrayHeaderSize + ((EvalMaxArgs) * (msgp.BytesPrefixSize + config.MaxLogicSigMaxSize))
```
We are increasing `s` in the first line by 15K, can't really imagine changing the second like in a rational way that reduces it by 15K.
In any case this value will depend on what we decide to do about [this](https://github.com/algorand/go-algorand/pull/6057/#issuecomment-2241226095)

Reply:
Following the implementation of this [comment](https://github.com/algorand/go-algorand/pull/6057#issuecomment-2255765284) we now have:
`const TxnTagMaxSize = 4394756`
which is close but not exactly the original `4620031`.

We could of course define a new constant equal to the original number and use that together with the msgp `maxtotalbytes` directive if we wanted, but I guess there is no reason to use a magic number.